apis/sql/v1beta1/sqlinstance_types.go

```diff
@@ -437,7 +437,36 @@ type InstanceValueFrom struct {
 	SecretKeyRef *v1alpha1.SecretKeyRef `json:"secretKeyRef,omitempty"`
 }
 
+type BinLogCoordinates struct {
+	// BinLogFileName: Name of the binary log file for a Cloud SQL instance.
+	BinLogFileName string `json:"binLogFileName,omitempty"`
+
+	// BinLogPosition: Position (offset) within the binary log file.
+	BinLogPosition int64 `json:"binLogPosition,omitempty,string"`
+}
+
+type CloneSource struct {
+	/* BinLogCoordinates: Binary log coordinates, if specified, identify the
+	position up to which the source instance is cloned. If not specified, the
+	source instance is cloned up to the most recent binary log coordinates. */
+	BinLogCoordinates *BinLogCoordinates `json:"binLogCoordinates,omitempty"`
+
+	/* DatabaseNames: (SQL Server only) Clone only the specified databases from the
+	source instance. Clone all databases if empty. */
+	DatabaseNames []string `json:"databaseNames,omitempty"`
+
+	/* PointInTime: Timestamp, if specified, identifies the time to which the
+	source instance is cloned. */
+	PointInTime string `json:"pointInTime,omitempty"`
+
+	/* SourceID: The ID of the source database instance to clone */
+	SourceID string `json:"sourceID,omitempty"`
```

Comment:
This looks like a reference to me. Can we reference a SQLInstance here?

Reply:
Good catch. Thanks. Fixed!